test/test_evaluator.cpp

```diff
@@ -295,6 +295,19 @@ TEST_F(ExprEvalTest, Array) {
   checkIntValue(evaluator, bb, 5L);
 }
 
+TEST_F(ExprEvalTest, EmptyArray) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  auto* a = IrBuilder::create<Val>(
```

Comment:
If a wrong array length is passed as input, e.g.
```
  auto* a = IrBuilder::create<Val>(
      std::vector<int64_t>{},
      ArrayType{std::make_shared<DataType>(DataType::Int), 2});
```
the err msg still shows `Empty array is not supported`.

Reply:
Fixed. I added a check about if the size matches.